UtopiaCity/Controllers/FireDepartment/FireDepartmentController.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using UtopiaCity.Services.FireDepartment.FireIncidentReportServices;
+using UtopiaCity.ViewModels.FireDepartment;
+
+namespace UtopiaCity.Controllers.FireDepartment
+{
+    public class FireDepartmentController : Controller
+    {
+        private readonly IFireIncidentReportService _fireIncidentReportService;
+
+        public FireDepartmentController(IFireIncidentReportService fireIncidentReportService)
+        {
+            _fireIncidentReportService = fireIncidentReportService;
+        }
+
+        public IActionResult Index()
+        {
+            var viewModel = _fireIncidentReportService.GetAllFireIncidentReportViewModels();
+            return View(viewModel);
+        }
+
+        [HttpGet]
+        public IActionResult CreateFireIncidentReport()
+        {
+            CreateFireIncidentReportViewModel viewModel = new CreateFireIncidentReportViewModel();
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        public IActionResult CreateFireIncidentReport(CreateFireIncidentReportViewModel viewModel)
+        {
+            try
+            {
+                _fireIncidentReportService.CreateFireIncidentReport(viewModel);
+                return RedirectToAction("Index");
+            }
+            catch (Exception)
+            {
+
```

Comment:
if exception is not hanlded properly, why "handle" it in first place?

Reply:
oh, sorry. I've forgotten to add return Bad Request